src/CSF.SRDashboard.Client/Pages/SeafarerProfile.razor.cs

```diff
@@ -84,19 +118,38 @@ protected async override Task OnInitializedAsync()
                     DownloadLink = link
                 });
             }
+
+            var uri = navigationManager.ToAbsoluteUri(navigationManager.Uri);
+            
+            if (QueryHelpers.ParseQuery(uri.Query).TryGetValue("requestId", out var _requestId))
+            {
+                RequestID = Convert.ToInt32(_requestId);
+            }
```

Comment:
Careful if the convert fails this will throw an exception. It will either throw a FormatException if requestId is not a valid Int32  or it will throw an OverflowException if the number  is greater than an Int32
```suggestion
            if (QueryHelpers.ParseQuery(uri.Query).TryGetValue("requestId", out var requestId))
            {
                RequestID = Convert.ToInt32(requestId);
            }
```

Reply:
Changed